### DISCUSSION TITLE
Support for RDF-star in rdflib

### DISCUSSION THREAD

**Discussion by akhil821100** [RDFLib/rdflib#1889] [General]
I and my team have created a module for support  RDF-star in rdflib. The module iterates through the embedded triples and converts them recursively into n-triple format. Moreover, it also has support for multiple graphs as well.

**Comment by akhil821100** [comment]
Additionally, our module also has support for quad-star and n-quad triples.

**Reply by IshanMehta115** [reply]
The program uses a recursive function `process_ntriple(n_triple_string, graph_name)` which takes in 2 input parameters, <br>
`n_triple_string` which is the n_triple in string format and <br>
`graph_name` which is the name of the graph in which the triple is to be added. <br>
The function then recursively calls itself on both the subject and object of the triple (if they are nested) and finally converts the nested n-triple into some number of non nested n-triples format.

**Comment by ghost** [comment]
There's an ongoing programme of existing RDFLib RDF Star work being done in https://github.com/RDFLib/rdflib-rdfstar
